libcxx/include/__functional/function.h

```diff
@@ -17,13 +17,7 @@
 #include <__functional/binary_function.h>
 #include <__functional/invoke.h>
 #include <__functional/unary_function.h>
-#include <__iterator/iterator_traits.h>
 #include <__memory/addressof.h>
-#include <__memory/allocator.h>
-#include <__memory/allocator_destructor.h>
-#include <__memory/allocator_traits.h>
-#include <__memory/compressed_pair.h>
-#include <__memory/unique_ptr.h>
 #include <__type_traits/aligned_storage.h>
```

Comment:
In the PR description, can you please describe the functional change that this introduces? I know you believe it's small and likely inconsequential, but `std::function` is one of the most widely used types in the library, so we're bound to hit Hyrum's law. I'd like a description of the impact of the change, as small as it is perceived, so that we can decide how to best roll it out. A release note also seems desirable.

Reply:
I couldn't find that it's a DR, so I've removed that claim. I still think we can treat it as a DR though.